python/traverse.py

```diff
@@ -26,50 +26,70 @@
 }
 
 
-def traverse_and_execute(
-    html: str, fn: Callable[[lxml.html.HtmlElement], None]
-) -> None:
-    elements = lxml.html.fragments_fromstring(html)
+def get_elements_list(html: str) -> List[slp.Node]:
+    parser = slp.HTMLParser(html)
+    res = []
 
-    # If there's leading text, the first element of the array will be a string.
-    # We can just discard that.
-    if isinstance(elements[0], str):
-        del elements[0]
+    if parser.head is not None:
+        res.extend(parser.head.iter(include_text=True))
```

Comment:
Same question as below, why are we looking at `head` at all?

Reply:
It's a quirk of the parser. When `HTMLParser(html)` is called on an HTML fragment, it treats the whole thing as it's own HTML page, putting some leading tags into the `head` automatically, and some into the `body` (which it creates). Checking for `None` and looking at the iter, we read into those tags and just get the parsed version of the input.